pkg/kubectl/cmd/exec/exec.go

```diff
@@ -84,6 +98,7 @@ func NewCmdExec(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.C
 			cmdutil.CheckErr(options.Run())
 		},
 	}
+	cmdutil.AddPodRunningTimeoutFlag(cmd, defaultPodExecTimeout)
 	cmd.Flags().StringVarP(&options.PodName, "pod", "p", options.PodName, "Pod name")
 	cmd.Flags().MarkDeprecated("pod", "This flag is deprecated and will be removed in future. Use exec POD_NAME instead.")
```

Comment:
Can you open a separate issue to remove this flag? I think it's marked deprecated for a couple of releases now so we should be able to remove it. It would simplify a lot of the code.

Reply:
do you mean `--pod` flag?
`--pod` flag was deprecated in July last year https://github.com/kubernetes/kubernetes/commit/888119a47cd and I think this should stay around until July next month. of course I am willing to open the issue to delete this flag in time

Reply:
Yeah, I complained before, but now 1.15 is a good time to remove the flag. Open a separate PR doing this, with the release date planned sometime in June we're good to remove it. Ping me in the PR please :smile:

Reply:
OK, I will open a new PR related to that :sweat_smile:

Reply:
@soltysh I have open separate PR to remove this flag.